x-pack/plugins/apm/server/routes/source_maps/route.ts

```diff
@@ -169,6 +192,85 @@ const uploadSourceMapRoute = createApmServerRoute({
   },
 });
 
+const uploadAndroidMapRoute = createApmServerRoute({
+  endpoint: 'POST /api/apm/androidmaps 2023-10-31',
+  options: {
+    tags: ['access:apm', 'access:apm_write'],
+    body: { accepts: ['multipart/form-data'], maxBytes: 100 * 1024 * 1024 },
+  },
+  params: t.type({
+    body: t.type({
+      service_name: t.string,
+      service_version: t.string,
+      map_file: t
+        .union([t.string, stringFromBufferRt])
+        .pipe(androidMapValidation),
+    }),
+  }),
+  handler: async ({
+    params,
+    plugins,
+    core,
+    logger,
+    featureFlags,
+  }): Promise<Artifact | undefined> => {
+    throwNotImplementedIfSourceMapNotAvailable(featureFlags);
+
+    const {
+      service_name: serviceName,
+      service_version: serviceVersion,
+      map_file: sourceMapContent,
+    } = params.body;
+    const bundleFilePath = 'android';
+    const fleetPluginStart = await plugins.fleet?.start();
+    const coreStart = await core.start();
+    const internalESClient = coreStart.elasticsearch.client.asInternalUser;
+    const savedObjectsClient = await getInternalSavedObjectsClient(coreStart);
+    try {
+      if (fleetPluginStart) {
+        // create source map as fleet artifact
+        const artifact = await createFleetAndroidMapArtifact({
+          fleetPluginStart,
+          apmArtifactBody: {
+            serviceName,
+            serviceVersion,
+            bundleFilepath: bundleFilePath,
+            sourceMap: sourceMapContent,
+          },
+        });
+
+        // sync source map to APM managed index
+        await createApmAndroidMap({
+          internalESClient,
+          logger,
+          fleetId: artifact.id,
+          created: artifact.created,
+          mapContent: sourceMapContent,
+          bundleFilepath: bundleFilePath.toString(),
```

Comment:
`bundleFilePath` is a string. Is this needed?

Reply:
You're right, I think it's not needed. I've just removed it 👍